swap/simulations_test.go

```diff
@@ -651,18 +676,45 @@ func TestBasicSwapSimulation(t *testing.T) {
 	log.Info("Simulation ended")
 }
 
-func waitForChequeProcessed(t *testing.T, ts *testService) error {
+func waitForChequeProcessed(t *testing.T, ts *testService, counter metrics.Counter, lastCount int64) error {
 	t.Helper()
 	ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
 	defer cancel()
 
-	backend := ts.swap.backend.(*swapTestBackend)
+	// we are going to wait for two things:
+	// * that a cheque has been processed
+	// * that the other side actually did process the testMsgPrice message
+	//   (by checking that the message counter has been increased)
+	var wg sync.WaitGroup
+	wg.Add(2)
+
+	go func() {
+		backend := ts.swap.backend.(*swapTestBackend)
+		select {
+		case <-ctx.Done():
+			t.Fatal("timed out waiting for cheque to be processed")
+		case <-backend.cashDone:
+			wg.Done()
+		}
+	}()
+
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				t.Fatal("Timed out waiting for all swap peer connections to be established")
```

Comment:
The message here seems wrong. Isn't this waiting for message to arrive?

Reply:
Oh hehe, well spotted, copy paste...